testutil/testfactory/types.go

```diff
@@ -0,0 +1,62 @@
+package testfactory
+
+import (
+	mrand "math/rand"
+	"sort"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	"github.com/tendermint/tendermint/types"
+)
+
+func GenerateRandomlySizedBlobs(count, maxBlobSize int) []types.Blob {
+	blobs := make([]types.Blob, count)
+	for i := 0; i < count; i++ {
+		blobs[i] = GenerateRandomBlob(tmrand.Intn(maxBlobSize))
+		if len(blobs[i].Data) == 0 {
+			i--
+		}
+	}
+
+	// this is just to let us use assert.Equal
+	if count == 0 {
+		blobs = nil
+	}
+
+	sort.Sort(types.BlobsByNamespace(blobs))
+	return blobs
+}
+
+func GenerateRandomBlob(size int) types.Blob {
+	blob := types.Blob{
+		NamespaceID:  tmrand.Bytes(appconsts.NamespaceSize),
```

Comment:
[no change needed] in the future, we may want this to generate a valid blob namespace. In other words, something like https://github.com/celestiaorg/celestia-app/blob/5bbdac2d3f46662a34b2111602b8f964d6e6fba5/testutil/namespace/random_ns_generator.go#L11-L23